s3/src/lib.rs

```diff
@@ -107,12 +107,21 @@ impl S3Provider {
         container: Container,
     ) -> Result<Vec<u8>, Box<dyn Error + Sync + Send>> {
         let rt = tokio::runtime::Runtime::new().unwrap();
-        rt.block_on(s3::remove_bucket(
-            &self.clients.read().unwrap()[actor],
-            &container.id,
-        ))?;
-
-        Ok(vec![])
+        serialize(
+            match rt.block_on(s3::remove_bucket(
```

Comment:
see previous comment on making this simpler code by replacing the match with a `map_or_else`